creme/tree/_nodes/base.py

```diff
@@ -435,26 +422,27 @@ def attribute_observers(self):
     def attribute_observers(self, attr_obs):
         self._attribute_observers = attr_obs       # noqa
 
-    def update_attribute_observers(self, X, y, weight, tree):
-        for idx, x in enumerate(X):
+    def update_attribute_observers(self, X, y, sample_weight, tree):
+        for idx, x in X.items():
             try:
                 obs = self.attribute_observers[idx]
             except KeyError:
-                if tree.nominal_attributes is not None and idx in tree.nominal_attributes:
+                if ((tree.nominal_attributes is not None and idx in tree.nominal_attributes)
+                        or not isinstance(x, numbers.Number)):
                     obs = self.new_nominal_attribute_observer()
                 else:
                     obs = self.new_numeric_attribute_observer()
                 self.attribute_observers[idx] = obs
-            obs.update(x, y, weight)
+            obs.update(x, y, sample_weight)
 
     def get_best_split_suggestions(self, criterion, tree):
-        """ Find possible split candidates.
+        """Find possible split candidates.
 
         Parameters
         ----------
-        criterion: SplitCriterion
+        criterion
             The splitting criterion to be used.
-        tree:
+        tree
             Hoeffding Tree.
```

Comment:
We can drop the "Hoeffding" part for generalization.

Reply:
Changed in commit [2a55a14](https://github.com/creme-ml/creme/pull/362/commits/2a55a14fa6f2b98d3160bd46f55f85d9fb6f272c).